oci_discovery/ref_engine_discovery/__init__.py

```diff
@@ -57,11 +86,11 @@ def resolve(engines, name):
                         else:
                             root['casEngines'] = []
                         root['casEngines'].extend(engine_reference.cas_engines)
+                    root_hash = hash(_json.dumps(root, sort_keys=True))
+                    if root_hash in roots:
+                        continue
+                    roots.add(root_hash)
                     yield root
             except Exception as error:
                 _LOGGER.warning(error)
-                raise
-                #continue
-            if not found_root:
```

Comment:
Need to drop the earlier `found_root` initialization too.

Reply:
> Need to drop the earlier `found_root` initialization too.

Done with 6e5b214 → 49c1d6e.  More details on why in the 49c1d6e commit message.